src/mlpack/tests/quantization_tests.cpp

```diff
@@ -0,0 +1,37 @@
+#include <mlpack/core.hpp>
+#include <mlpack/methods/ann/layer/layer.hpp>
+#include <mlpack/methods/ann/layer/linear.hpp>
+#include <mlpack/methods/ann/ffn.hpp>
+#include <mlpack/methods/ann/quantization/quantization_utils.hpp>
+
+#include "catch.hpp"
+
+using namespace mlpack;
+using namespace mlpack::ann;
+using namespace arma;
+
+TEST_CASE("QuantizeLinearLayerTest", "[QuantizationTest]")
+{
+    LinearType<> layer(5, NoRegularizer());
+    layer.Parameters().randu();
+
+    LinearQuantization quantizer;
+    arma::imat quantizedWeights = quantizer.QuantizeWeights(layer.Parameters());
+
+    REQUIRE(quantizedWeights.min() >= -127);
+    REQUIRE(quantizedWeights.max() <= 127);
+}
+
+TEST_CASE("CloneLayerTest", "[QuantizationTest]")
+{
+    LinearType<arma::mat, NoRegularizer> layer(4, NoRegularizer());
+    layer.Parameters().randu();
+
+    auto clonedLayer = layer.CloneAs<arma::fmat>();
+
+    REQUIRE(clonedLayer->Parameters().n_rows == layer.Parameters().n_rows);
+    REQUIRE(clonedLayer->Parameters().n_cols == layer.Parameters().n_cols);
+    REQUIRE(clonedLayer->Parameters().is_finite());
+
+    delete clonedLayer;
+}
```

Comment:
Some other tests I would suggest adding:

 * Direct tests on the quantization strategy functions to ensure the output matrices have the right size, range, this type of thing.
 * Tests that setting `numBits` to something different make a difference in the output.  (I don't have a complete understanding of `numBits` at this point.)
 * If you want to test the `Clone()` or `CloneAs()` function of each layer, that would actually probably be better done by adding a test to each layer's test file.  But, given the simplicity of each layer's copy constructors, I am okay if they are only tested as a larger part of the quantization tests and not individually.